runtime/src/actor_error.rs

```diff
@@ -14,41 +14,81 @@ pub struct ActorError {
     exit_code: ExitCode,
     /// Message for debugging purposes,
     msg: String,
+    /// Optional exit data
+    data: Option<IpldBlock>,
 }
 
 impl ActorError {
     /// Creates a new ActorError. This method does not check that the code is in the
     /// range of valid actor abort codes.
     pub fn unchecked(code: ExitCode, msg: String) -> Self {
-        Self { exit_code: code, msg }
+        Self { exit_code: code, msg, data: None }
+    }
+
+    pub fn unchecked_with_data(code: ExitCode, msg: String, data: Option<IpldBlock>) -> Self {
+        Self { exit_code: code, msg, data }
+    }
+
+    /// Creates a new ActorError. This method checks if the exit code is within the allowed range,
+    /// and automatically converts it into a user code.
+    pub fn checked(code: ExitCode, msg: String) -> Self {
```

Comment:
I recognize this code as being copied (twice) from `runtime/fvm.rs`. I don't think it belongs here. It's appropriate only in the case of receiving an exit code from a `send`, not in various other places that exit codes propagate.

If you factored this match block out from the runtime as a helper function, `adapt_callee_code` or similar, that might be ok. But who's going to call it except the runtime? Links to how this is used might help.

Reply:
@anorth Good call, thank you. I'm going to refactor runtime/fvm.rs to use _this_ method, instead, if that works for you. I'm also dropping the repeated copy -- just one `checked` method is sufficient, I think.